.github/workflows/gitlab-ci.yml

```
@@ -0,0 +1,22 @@
+# Copyright 2022 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Author: Sergio Mazzola <smazzola@iis.ee.ethz.ch>
+
+name: gitlab-ci
+
+on: [ push, pull_request, workflow_dispatch ]
```

While I am still learning myself, I think it would be valuable to differentiate the triggers: `pull_request` makes sense, but is unnecessary if it is running on `push`, as `push` will trigger on all commits. I would suggest to limiting to push to the main branch, and pull request: this way, it won't run on all commits in the repo (unless you want to with workflow_dispatch).
```suggestion
on:
  push:
    branches:
      - main
  pull_request:
  workflow_dispatch:
```